seatunnel-plugin-discovery/src/main/java/org/apache/seatunnel/plugin/discovery/PluginDiscovery.java

```diff
@@ -78,4 +81,27 @@ Optional<T> createOptionalPluginInstance(
      * @return plugin instances.
      */
     List<T> getAllPlugins(List<PluginIdentifier> pluginIdentifiers);
+
+    /**
+     * Get all plugins(connectors and transforms)
+     *
+     * @return
+     */
+    default LinkedHashMap<PluginIdentifier, OptionRule> getPlugins() {
+        throw new UnsupportedOperationException("Not implemented");
+    }
+
+    /** Print all supported plugins(connectors and transforms) */
+    default void printSupportedPlugins() {
+        throw new UnsupportedOperationException("Not implemented");
+    }
+
+    /**
+     * Print option rules of the plugin by the plugin identifier
+     *
+     * @param pluginIdentifier
+     */
+    default void printOptionRules(String pluginIdentifier) {
+        throw new UnsupportedOperationException("Not implemented");
+    }
```

Comment:
It is strange to add the printing method in the `PluginDiscovery`. It is recommended to move the printing-related logic to the Command, and the `PluginDiscovery` returns the required information.

Reply:
> It is strange to add the printing method in the `PluginDiscovery`. It is recommended to move the printing-related logic to the Command, and the `PluginDiscovery` returns the required information.

OK